src/evict/evict_lru.c

```diff
@@ -2721,15 +2720,24 @@ __wt_verbose_dump_cache(WT_SESSION_IMPL *session)
      * Apply the overhead percentage so our total bytes are comparable with the tracked value.
      */
     total_bytes = __wt_cache_bytes_plus_overhead(conn->cache, total_bytes);
-
-    WT_RET(
-      __wt_msg(session, "cache dump: total found: %" PRIu64 "MB vs tracked inuse %" PRIu64 "MB",
-        total_bytes / WT_MEGABYTE, cache->bytes_inmem / WT_MEGABYTE));
-    WT_RET(__wt_msg(session, "total dirty bytes: %" PRIu64 "MB vs tracked dirty %" PRIu64 "MB",
-      total_dirty_bytes / WT_MEGABYTE,
-      (cache->bytes_dirty_intl + cache->bytes_dirty_leaf) / WT_MEGABYTE));
-    WT_RET(__wt_msg(session, "total updates bytes: %" PRIu64 "MB vs tracked updates %" PRIu64 "MB",
-      total_updates_bytes / WT_MEGABYTE, __wt_cache_bytes_updates(cache) / WT_MEGABYTE));
+    cache_bytes_updates = __wt_cache_bytes_updates(cache);
+
+    WT_RET(__wt_msg(session,
+      "cache dump: total found: %.2f"
+      "MB vs tracked inuse %.2f"
+      "MB",
+      (double)total_bytes / WT_MEGABYTE, (double)cache->bytes_inmem / WT_MEGABYTE));
+    WT_RET(__wt_msg(session,
+      "total dirty bytes: %.2f"
+      "MB vs tracked dirty %.2f"
+      "MB",
+      (double)total_dirty_bytes / WT_MEGABYTE,
+      (double)(cache->bytes_dirty_intl + cache->bytes_dirty_leaf) / WT_MEGABYTE));
+    WT_RET(__wt_msg(session,
+      "total updates bytes: %.2f"
+      "MB vs tracked updates %.2f"
+      "MB",
+      (double)total_updates_bytes / WT_MEGABYTE, (double)cache_bytes_updates / WT_MEGABYTE));
```

Comment:
Was there a reason we needed to save `cache_bytes_updates` into a local var instead of using `(double)__wt_cache_bytes_updates(cache) / WT_MEGABYTE`?

Reply:
Directly typecasting the function return will result in the compiler complaining about casting a function call to non-matching type "double". Here, casting to a "double" is done in order to avoid truncating the division result and to allow the result to be printed to 2dp. Since we're tracking bytes, type "double" should be sufficient to avoid data loss in conversion.